### PR TITLE
Stricter email validation

### PR DESCRIPTION
**Pull request by gonzaloriestra**
Use [email_address gem](https://github.com/afair/email_address) to avoid invalid email domains.

Related to: https://github.com/CartoDB/cartodb-central/issues/2533

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -159,31 +160,47 @@ def self.new_with_organization(organization, viewer: false)
   ## Validations
   def validate
     super
+    validate_username
+    validate_email
+    validate_password
+    validate_organization
+    validate_quotas
+  end
```
Comment (alrocar) [inline app/models/user.rb]: ❤️

File: config/initializers/email_address.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+EmailAddress::Config.configure(local_format: :conventional)
+
+class EmailValidator < ActiveModel::EachValidator
```
Comment (alrocar) [inline config/initializers/email_address.rb]: we could add the validation to the AR model as well

Comment (gonzaloriestra) [reply config/initializers/email_address.rb]: I thought about it, but it doesn't have any kind of validation yet... Do you think it's worth it? Or should I add all of them?

Comment (alrocar) [reply config/initializers/email_address.rb]: Oh I see... then nothing

File: app/models/user.rb [hunk 3]
```diff
@@ -159,31 +160,47 @@ def self.new_with_organization(organization, viewer: false)
...
     validates_max_length 63, :username
     errors.add(:name, 'is taken') if name_exists_in_organizations?
+  end
 
+  def validate_email
     validates_presence :email
-    validates_unique   :email, :message => 'is already taken'
-    validates_format EmailAddressValidator::Regexp::ADDR_SPEC, :email, :message => 'is not a valid address'
+    validates_unique   :email, message: 'is already taken'
+    errors.add(:email, EmailAddress.error(email)) unless EmailAddress.valid?(email)
```
Comment (ethervoid) [inline app/models/user.rb]: We have to include this validation in the activerecord model as well

Comment (gonzaloriestra) [reply app/models/user.rb]: https://github.com/CartoDB/cartodb/pull/15030#discussion_r305316953

**Pull request merged**